js/pulse.js

```diff
@@ -431,21 +459,63 @@
             }
 
 
+            function onOverlayFinished(){
+                clearTimeout(overlayTimer);
+                amc.notifyNonlinearAdEnded(currentOverlayAd.id);
+                currentOverlayAd = null;
+            }
+
+            function startOverlayCountdown(){
+                lastOverlayAdStart = Date.now();
+                overlayTimer = setTimeout(onOverlayFinished, overlayTimeLeftMillis);
+            }
+
+            function onOverlayShown(){
+                if(currentOverlayAd){
+                    overlayTimeLeftMillis = currentOverlayAd.ad.getDuration() * 1000;
+                    adPlayer.overlayAdShown(currentOverlayAd.ad);
+                    startOverlayCountdown();
+                }
+            }
+
+            function overlayPause(){
+                if(currentOverlayAd){
+                    overlayTimeLeftMillis = overlayTimeLeftMillis - (Date.now() - lastOverlayAdStart);
+                    clearTimeout(overlayTimer);
+                }
+            }
+
             /**
              * Mandatory method. Called by the AMF when an ad play has been requested
              * @param v4ad
              */
             this.playAd = function(v4ad) {
-
                 //If the SDK is not loaded, tell the AMC our placeholder ad is finished
                 if(!adModuleJsReady){
                     amc.notifyPodEnded(v4ad.id);
                     return;
                 }
 
-                podStarted = v4ad.id;
+                if(v4ad.adType === amc.ADTYPE.NONLINEAR_OVERLAY){
+                    if(contentPaused){
+                        currentPauseAd = v4ad;
+                    } else {
+                        currentOverlayAd = v4ad;
+                    }
+
+                    amc.sendURLToLoadAndPlayNonLinearAd(v4ad.ad, v4ad.id, v4ad.ad.getResourceURL());
+
+                    //Assume the ad was loaded
+                    if(!contentPaused){
```

Comment:
Will the case where the content is paused and seeked forward ahead of the ad breakpoint to trigger the overlay work fine? Just wanted to double check.